core/src/components/datetime/datetime.tsx

```diff
@@ -2113,6 +2157,15 @@ export class Datetime implements ComponentInterface {
               }
             }
 
+            /**
+             * Some days are disabled through max & min or allowed dates
+             * and also by the component being readonly or disabled.
+             * These need to be displayed differently.
+             */
+            const isCalDayDoublyDisabled = isCalDayDisabled && isDatetimeDisabled;
```

Comment:
I'd rename this back to `isCalDayConstrained`, then this PR is good to merge. 

"doubly disabled" is confusing in my opinion. An interactive can only be in a singular disabled state. 

Having the variable match the CSS class will probably be best for now, giving queues to both our dev team and 3rd party contributors to where they can look for the associated implementation.